java-client/barrage/src/main/java/io/deephaven/client/impl/BarrageSubscriptionImpl.java

```diff
@@ -263,8 +254,13 @@ private void cancel() {
             return;
         }
 
-        resultTable.handleBarrageError(new RequestCancelledException("Barrage subscription cancelled by client"));
-        GrpcUtil.safelyCancel(observer, "Barrage subscription is cancelled", null);
+        final RequestCancelledException cancelledException =
```

Comment:
We're missing a future.completeExceptionally in the destroy case. But the future might be null. How dystopian.